build-system/compile/compile.js

```diff
@@ -396,19 +398,20 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
           })
           .pipe(nop())
           .on('end', resolve);
-    } else {
-      return gulp.src(srcs, {base: '.'})
-          .pipe(gulpIf(shouldShortenLicense, shortenLicense()))
-          .pipe(sourcemaps.init({loadMaps: true}))
-          .pipe(gulpClosureCompile(compilerOptionsArray))
-          .on('error', err => {
-            handleCompilerError(outputFilename);
-            reject(err);
-          })
-          .pipe(rename(outputFilename))
-          .pipe(sourcemaps.write('.'))
-          .pipe(gulp.dest(outputDir))
-          .on('end', resolve);
     }
+
+    return gulp.src(srcs, {base: '.'})
+        .pipe(gulpIf(shouldShortenLicense, shortenLicense()))
+        .pipe(sourcemaps.init({loadMaps: true}))
+        .pipe(gulpIf(({path}) => !/third_party/.test(path), babel({plugins})))
```

Comment:
`gulp-if` takes a regex as its first arg. For example:

https://github.com/ampproject/amphtml/blob/e9acfc63ccde5afe4d048d275b4320a771443396/build-system/compile/single-pass.js#L609

So this can also be written as a negative lookahead regex.
```
.pipe(gulpIf(/^(?!.*third_party)/, babel({plugins})))
```